itest/lnd_coop_close_external_delivery_test.go

```diff
@@ -62,17 +86,52 @@ func testCoopCloseWithExternalDelivery(ht *lntest.HarnessTest) {
 // balance irrespective of whether the delivery address is in LND's wallet or
 // not. Some users set this value to be an address in a different wallet and
 // this should not affect our ability to accurately report the settled balance.
+//
+// If importTapscript is set, it imports a Taproot script and internal key to
+// alice LND to make sure it doesn't interfere with channel delivery address.
+// See https://github.com/lightninglabs/loop/issues/923
 func testCoopCloseWithExternalDeliveryImpl(ht *lntest.HarnessTest,
-	alice, bob *node.HarnessNode, upfrontShutdown bool,
+	alice, bob *node.HarnessNode, upfrontShutdown, importTapscript bool,
 	deliveryAddressType lnrpc.AddressType) {
 
-	// Here we generate a final delivery address in bob's wallet but set by
-	// alice. We do this to ensure that the address is not in alice's LND
-	// wallet. We already correctly track settled balances when the address
-	// is in the LND wallet.
-	addr := bob.RPC.NewAddress(&lnrpc.NewAddressRequest{
-		Type: deliveryAddressType,
-	})
+	var addr string
+	if importTapscript {
+		// Make fake taproot internal public key (equal to the final).
+		taprootPubkey := [32]byte{1, 2, 3}
+		if upfrontShutdown {
+			// Make new address for second sub-test not to import
+			// the same address twice causing an error.
+			taprootPubkey[3] = 1
+		}
+		pkScriptBytes := append(
+			[]byte{txscript.OP_1, txscript.OP_DATA_32},
+			taprootPubkey[:]...,
+		)
+		pkScript, err := txscript.ParsePkScript(pkScriptBytes)
+		require.NoError(ht, err)
+		address, err := pkScript.Address(harnessNetParams)
+		require.NoError(ht, err)
+		addr = address.String()
+
+		// Import the taproot address to LND.
+		req := &walletrpc.ImportTapscriptRequest{
+			InternalPublicKey: taprootPubkey[:],
+			Script: &walletrpc.ImportTapscriptRequest_FullKeyOnly{
+				FullKeyOnly: true,
+			},
+		}
+		res := alice.RPC.ImportTapscript(req)
```

Comment:
Do keys imported via `ImportPublicKey` also have this issue? I think not since the internal keys are unknown to `lnd`?

Reply:
I added more test cases for "coop close with external delivery" to safeguard against this in the future:
  - P2TR address imported with ImportPublicKey
  - P2WPKH address imported with ImportPublicKey